src/components/Swap/Toolbar/index.tsx

```diff
@@ -94,10 +117,52 @@ export default memo(function Toolbar() {
     inputAmount,
     isWrap,
     trade,
-    outputUSDC,
     impact,
+    open,
+    onExpand,
+    outputUSDC,
   ])
 
+  const tradeSummaryRows: SummaryRowProps[] = useMemo(() => {
+    const currencySymbol = trade?.outputAmount?.currency.symbol ?? ''
+    const rows: SummaryRowProps[] = []
+    rows.push({
+      name: t`Network fee`,
+      value: gasUseEstimateUSD ? `~${formatCurrencyAmount(gasUseEstimateUSD, NumberType.FiatGasPrice)}` : '-',
+    })
+    rows.push({
+      color: impact?.warning,
+      name: t`Price impact`,
+      value: impact?.percent ? formatPriceImpact(impact?.percent) : '-',
+      valueTooltip: impact?.warning
+        ? {
+            icon: AlertTriangle,
+            content: <Caption.PriceImpactWarningTooltipContent />,
+          }
+        : undefined,
+    })
+    rows.push({
+      name: t`Minimum output after slippage`,
+      value: trade ? `${formatCurrencyAmount(trade?.minimumAmountOut(slippage.allowed))} ${currencySymbol}` : '-',
+    })
+    rows.push({
+      name: t`Expected output`,
+      value: trade ? `${formatCurrencyAmount(trade?.outputAmount)} ${currencySymbol}` : '-',
+      nameTooltip: trade
+        ? {
+            icon: Info,
+            content: <SwapInputOutputEstimate trade={trade} slippage={slippage} />,
+          }
+        : undefined,
+    })
+    return rows
+  }, [gasUseEstimateUSD, impact?.percent, impact?.warning, slippage, trade])
+
+  const expandedHeight = useMemo(() => {
```

Comment:
can you leave a comment above this one to explain what the logic for calculating height is? my brain turns off when seeing this code lol